packages/hawtio/src/plugins/camel/routes/CamelRoutes.tsx

```diff
@@ -0,0 +1,313 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import { CamelRoute, routesService } from '@hawtiosrc/plugins/camel/routes-service'
+import { Caption, TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
+import {
+  Button,
+  Card,
+  CardBody,
+  Dropdown,
+  DropdownItem,
+  KebabToggle,
+  Modal,
+  ModalVariant,
+  Page,
+  Text,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+} from '@patternfly/react-core'
+import { AsleepIcon, InfoCircleIcon, PlayIcon, Remove2Icon } from '@patternfly/react-icons'
+import { eventService } from '@hawtiosrc/core'
+import { workspace } from '@hawtiosrc/plugins/shared'
+
+export const CamelRoutes: React.FunctionComponent = () => {
+  const { selectedNode } = useContext(CamelContext)
+  const [routes, setRoutes] = useState<CamelRoute[]>([])
+  const [selected, setSelected] = useState<Map<string, boolean>>(new Map())
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isConfirmDeleteOpen, setIsConfirmDeleteOpen] = useState(false)
+  const getRouteAttributes = () => {
+    routesService.getRoutesAttributes(selectedNode).then(routes => {
+      const tmpSelected = new Map(selected)
+      routes.forEach(route => {
+        selected.set(route.RouteId, selected.get(route.RouteId) ?? false)
+      })
+      setSelected(tmpSelected)
+      setRoutes(routes)
+    })
+  }
+  useEffect(() => {
+    getRouteAttributes()
+  }, [])
+
+  const onDropdownToggle = (isOpen: boolean) => {
+    setIsOpen(isOpen)
+  }
+
+  const onSelect = (routeId: string, isSelecting: boolean) => {
+    selected.set(routeId, isSelecting)
+    setSelected(new Map(selected))
+  }
+
+  const onCheckAll = (isSelected: boolean) => {
+    const tmpChecked = new Map<string, boolean>()
+    selected.forEach((_v, key: string) => {
+      tmpChecked.set(key, isSelected)
+    })
+    setSelected(tmpChecked)
+  }
+
+  const isAllSelected = (): boolean => {
+    let res = true
+    selected.forEach(value => {
+      if (!value) res = value
+    })
+    return res
+  }
+
+  const isSuspendEnabled = (state: string): boolean => {
+    let res = false
+    selected.forEach((value, key) => {
+      const route = routes.find(r => r.RouteId === key)
+      if (value && route && route.State === state) res = value
+    })
+    return res
+  }
+
+  function onStopClicked() {
+    let stoppedCount = 0
+    selected.forEach(async (value, key) => {
+      if (value) {
+        const route = routes.find(r => r.RouteId === key)
+        if (route && route.State === 'Started') {
+          try {
+            await routesService.stopRoute(route.objectName as string)
+            stoppedCount++
+            getRouteAttributes()
```

Comment:
With this probably being the reason why `getRouteAttributes()` is outside the `useEffect`, may I suggest in this case that a flag is adopted instead that triggers the `useEffect` to run again rather than calling the function directly?